### ISSUE TITLE
this.subject() throws error in moduleForModel if { integration: true }

### ISSUE THREAD

**Issue by mitchlloyd** [switchfly/ember-test-helpers#46]
I have a test where I was passing `{ integration: true }` in order to use a model with a hasMany relationship. After upgrading from 0.3.1 to 0.3.2, running `this.subject()` throws an error:

```
TypeError: Cannot read property 'createRecord' of undefined
```

I've found that when setting the integration flag, `this.container.lookup('store:main')` returns `undefined`. Without the flag, `this.container.lookup('store:main')` returns a store object.

**Comment by rwjblue** [comment]
This is because `modelForModel` with `integration: true` is not running the initializers (or the setup code from `isolatedContainer`).

I think we should be adding [all of these registrations](https://github.com/switchfly/ember-test-helpers/blob/0a514b2367c19cf6f67728629617b63c55128dfc/lib/ember-test-helpers/isolated-container.js#L57-L94) from [inside the `_setupIntegrationContainer](https://github.com/switchfly/ember-test-helpers/blob/0a514b2367c19cf6f67728629617b63c55128dfc/lib/ember-test-helpers/test-module.js#L222-L238) in addition to calling `Ember.Application.buildRegistry`.

**Comment by rwjblue** [comment]
We should extract those registrations into a stand alone function, then call it from both `isolatedContainer` and `_setupIntegrationContainer`.

**Comment by mitchlloyd** [comment]
Thanks for the quick reply! For now I've got a workaround using `DS._setupContainer(this.container)`.

**Comment by wagenet** [comment]
@rwjblue you also said this was the same as the "missing component:-text-area" stuff, right?

**Comment by rwjblue** [comment]
Yes

**Comment by nikz** [comment]
Added a failing test for this in #51

**Comment by mitchlloyd** [comment]
Does it seem reasonable that Ember and Ember Data could maintain this list of registrations?

For instance, in Ember, this minimal set of container items could be extracted from the [application `buildRegistry` method](https://github.com/emberjs/ember.js/blob/257c2bc8dcc4c6f418abf6b1142c688d9dcee0c7/packages/ember-application/lib/system/application.js#L993) and be exposed as a public API.

I know that is beyond the scope of this bug, but I just wanted to see if this was a decent idea.

**Comment by rwjblue** [comment]
@mitchlloyd - Both Ember Data (in the last couple versions) and Ember (since 1.0.0!) have a mechanism for populating a container with what is needed for a default setup.  Unfortunately, we haven't been using them here.  I submitted #49 to resolve that, and remove this super annoying hard-coding of registrations/options/etc.

**Comment by rwjblue** [comment]
Resolved and released in ember-test-helpers@0.5.0 and ember-qunit@0.4.0.
